### DISCUSSION TITLE
When creating a mountpoint inside a container for the host changes are nor propagated

### DISCUSSION THREAD

**Discussion by TheClonerx** [juicedata/juicefs#3210] [Q&A]
I was trying to mount a juicefs client inside a container with the goal of having the mountpoint accessible from the host, however the changes made in a client are not being propagated to other clients in different mountpoints.

To reproduce:
* Start an object and metatada service locally, formatting them for juicefs: `(redis-server& minio server object-storage& sleep 1; juicefs format --storage minio --bucket minio://localhost:9000/data --access-key minioadmin --secret-key minioadmin redis://localhost/1 test& wait;)`
* Create a mountpoint in the host using a container: `podman run --network host -it --rm --device /dev/fuse --cap-add SYS_ADMIN -v /tmp:/mnt:rw docker.io/juicedata/mount juicefs mount redis://localhost/1 /mnt/test`
* Create a second mountpoint similar to the first one (using /mnt/test2 instead)
Now these two mountpoints are created under the host as `/tmp/test` & `/tmp/test2`
If you create a file in the first one, it will not appear in the second one:
* `echo 'Hello World!' > /tmp/test/greet.txt`
* `ls /tmp/test2` (no output)

**Comment by coutland** [comment]
According to the [juicefs documentation](https://juicefs.com/docs/community/juicefs_on_docker) you need to specify "rshared" on the bind mount. This specifies the "bind propagation" of the mount and causes nested mounts inside the container to propagate out to the host. So change the mount command to `podman run --network host -it --rm --device /dev/fuse --cap-add SYS_ADMIN -v /tmp:/mnt:rw,rshared docker.io/juicedata/mount juicefs mount redis://localhost/1 /mnt/test`

**Reply by TheClonerx** [reply]
Even after specifying the the `rshared` option I still observe the described behavior.
